### PR TITLE
Add pm-cpu to maint-1.0

### PR DESCRIPTION
**Pull request by ndkeen**
Add pm-cpu machine to maint-1.0. However, there are still a few changes required to use it out-of-the-box that are explained in "notes.txt" file. This PR is to get us closer and then resolve remaining issues.

Until we resolve the those issues, this PR does not completely fix https://github.com/E3SM-Project/E3SM/issues/5280

### CODE REVIEW

File: notes.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+This branch adds pm-cpu to maint-1.0, but there are still some changes
+required to get tests running.
+
+There are 2 changes in files under submodules, so will have to work out best way to fix.
+
+These are changes needed with higher version of GNU (which is all that is avail on PM)
+
+1) Need a change here to build. this is already done in master, but there are other changes we don't need --
+can just update source with the interface changes needed, by doing this:
```
Comment (rljacob) [inline notes.txt:9]: Did you already do this?  That file is modified in this PR.

File: notes.txt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+cp ODEMod.F90 components/clm/src/external_models/sbetr/src/betr/betr_math/ODEMod.F90
+
+2) Needed to work-around issue with solver during runtime. This change modifies compiler
+flags for ALL MPAS sources which is not necessary, but unable to only affect one source in MPAS
+In current master, we build one MPAS source file with a compiler flag to turn off very specific optimization,
+however, in maint-1.0, the CMake logic is different and I can't do the same thing in Depends.
+But what I can do is add a flag here (which may need more guards) that turns off the optimization
+for all MPAS source files.
+
+cp Makefile.in.ACME components/mpas-source/src/Makefile.in.ACME
```
Comment (rljacob) [inline notes.txt:20]: That's fine.  If we need to do something better, it can be a future PR.

File: notes.txt [hunk 3]
```diff
@@ -0,0 +1,31 @@
+This branch adds pm-cpu to maint-1.0, but there are still some changes
+required to get tests running.
+
+There are 2 changes in files under submodules, so will have to work out best way to fix.
+
+These are changes needed with higher version of GNU (which is all that is avail on PM)
```
Comment (rljacob) [inline notes.txt:6]: This implies the changes below haven't been done but it looks like some of them have.  Rewrite to describe what was already done vs. what still needs to be done.

**Review by rljacob — APPROVED** [review]

**Comment by rljacob** [comment]
See my comments on notes.txt

**Comment by rljacob** [comment]
The process for making submodule changes:   make a branch in the repo that has been submoduled and add the fix.  Push that branch to the repo we are pulling the submodule from.   Update the submodule to point to the head of that branch.  Repeat for each submodule that needs a fix.

If you can't push a branch to the submoduled repo, we will fork the submodule to E3SM-Project and change maint-1.0 to point to it.

**Comment by rljacob** [comment]
@ndkeen what is still holding up this PR?   We need this working at least a week before cori goes away.

**Comment by ndkeen** [comment]
I was just working on this. I have a new branch `ndk/maint-1.0/add-pm-cpu4` and make a new PR.

We still have unresolved issues -- one needs a change in MPAS submodule for intel builds and the other needs a manual change (also in MPAS submodule) as a work-around for certain GNU cases.

**Comment by ndkeen** [comment]
Closing in favor of https://github.com/E3SM-Project/E3SM/pull/5696

**Pull request closed without merging**